src/Symfony/Component/Security/Http/Tests/Firewall/SessionExpirationListenerTest.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Tests\Firewall;
+
+use Symfony\Component\Security\Http\Firewall\SessionExpirationListener;
+
+/**
+ * @author Antonio J. García Lagar <aj@garcialagar.es>
+ */
+class SessionExpirationListenerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testHandleWhenNoSession()
+    {
+        $request = $this->getMock('Symfony\Component\HttpFoundation\Request');
+        $request
+            ->expects($this->once())
+            ->method('getSession')
+            ->will($this->returnValue(false));
```

Comment:
this is wrong. the code returns either a SessionInterface instance or ``null``. It never returns ``false``.